### DISCUSSION TITLE
Seeing +4 hour delays on reacting to checkboxes in private org repo

### DISCUSSION THREAD

**Discussion by voxpelli** [renovatebot/renovate#11645] [Request Help]
I've used Renovate extensively for my own personal opensource projects ([my config](https://github.com/voxpelli/renovate-config-voxpelli), [a repo](https://github.com/voxpelli/node-connect-pg-simple)) and there when I'm checking a checkbox Renovate picks it up almost immediately, but now in an organization I'm in it takes way longer.

Since we added it it has always been slow, and right now I'm seeing a +4 hour delay on it reacting, see screenshot below.

What can be the cause of this? Is there something throttling the repo in Renovate? Is there some way for me to be able to debug this?

![Skärmavbild 2021-09-08 kl  15 27 58](https://user-images.githubusercontent.com/34457/132518298-d4694753-d7ab-4784-b01c-8f7a790800f9.png)

The https://app.renovatebot.com/ dashboard shows me this:

![Skärmavbild 2021-09-08 kl  15 29 46](https://user-images.githubusercontent.com/34457/132518669-8a574a86-16e1-4933-987f-559ac28efa5a.png)

**Comment by voxpelli** [comment]
For reference, this is the delay I get when I now did the exact same thing in the open here: https://github.com/voxpelli/node-connect-pg-simple/pull/204

30 seconds delay rather than 4 hours:

![Skärmavbild 2021-09-08 kl  16 03 10](https://user-images.githubusercontent.com/34457/132524303-a93ee24f-5d24-45e1-a8df-cae9a30a9e0d.png)

And internal PR is snow up at over 5 hours, so it's not like there was an outage in the hosted service:

![Skärmavbild 2021-09-08 kl  16 04 37](https://user-images.githubusercontent.com/34457/132524543-201c6d5d-bbf4-49fa-a65f-de0eb3cc3469.png)

**Reply by voxpelli** [reply]
A colleague tried on another internal repo of ours and there they had no delays like ours, so I guess it's something with our specific repo or setup that is causing it :/

Screenshot from colleague:

![Skärmavbild 2021-09-08 kl  16 17 06](https://user-images.githubusercontent.com/34457/132526663-69e9d576-441a-4c4f-ac52-3207aa2e5eee.png)

**Comment by rarkins** [comment]
Here's the raw log files from the S3 bucket:

![image](https://user-images.githubusercontent.com/6311784/132527012-ce75e4b5-a7b1-46d7-ba97-884d5dc124d6.png)

Here's from the job_log table in the DB:

![image](https://user-images.githubusercontent.com/6311784/132527437-41287085-5296-49bd-9742-cadd3cd45261.png)

The temporary error ones were both due to bad luck that an `npm` command was being run at the time a SIGTERM was received. We catch and block SIGTERMs in the Renovate core logic and wait for the current job to complete however the "exec" containers receive the SIGTERM independently and unfortunately return an exec error. So it's unfortunately but expected.

The job before that seemed to have just disappeared/timed out. I'd recommend you uncheck, pause a few seconds, then re-check the box again one more time

**Reply by voxpelli** [reply]
Thanks!

Would be very nice to see the running jobs in the dashboard as well, I can now see the `temporary error` ones, but not the other three:
![Skärmavbild 2021-09-08 kl  16 29 47](https://user-images.githubusercontent.com/34457/132529607-30c1df5f-599f-4e9f-bd8d-6629f302e26f.png)

Would be nice to be able to see that the toggle have picked up and see if eg. something in the running job is causing some unexpected inf-loops or something

**Reply by voxpelli** [reply]
Did what you suggested + let it be 10 minutes rather than a few seconds. Still no response in 30 minutes.

Maybe something is wrong with our config and/or other setup? I can share more in a private channel.

![Skärmavbild 2021-09-08 kl  17 16 41](https://user-images.githubusercontent.com/34457/132536969-8c57c55e-1836-47d7-a888-fc461fb55469.png)

**Reply by voxpelli** [reply]
Now it seems to be running more smoothly, but I looked at the timings and noticed that a run on our repo takes around 18 minutes apparently 😝 

I guess this would become vastly quicker if we move to use npm workspaces for our monorepo instead of having like 20-25 individual lock files? (I have added that to our backlog)

![Skärmavbild 2021-09-09 kl  10 11 13](https://user-images.githubusercontent.com/34457/132648731-276476d8-70c0-470f-856a-5087a8547b9e.png)

**Reply by viceice** [reply]
Yes, a mono repo with a single lockfile would speeup this a lot
